executable/microsoft_pe.ksy

```diff
@@ -408,3 +413,134 @@ types:
         size: length - 8
         doc: Contains a certificate, such as an Authenticode signature.
     doc-ref: 'https://docs.microsoft.com/en-us/windows/desktop/debug/pe-format#the-attribute-certificate-table-image-only'
+
+  resource_directory_table:
+    params:
+      - id: depth
+        type: u4
+      - id: section_file_offset
+        type: u4
+      - id: section_virtual_address
+        type: u4
+    seq:
+      # This field is reserved for future use. It is currently set to zero.
+      - id: characteristics
+        type: u4
+      - id: time_date_stamp
+        type: u4
+        -orig-id: 'TimeDateStamp'
+      - id: version
+        type: version_u2
+      - id: num_named_entries
+        type: u2
+        -orig-id: NumberOfNamedEntries
+      - id: num_id_entries
+        type: u2
+        -orig-id: NumberOfIdEntries
+      - id: named_entries
+        type: resource_directory_entry(depth)
+        repeat: expr
+        repeat-expr: num_named_entries
+      - id: id_entries
+        type: resource_directory_entry(depth)
+        repeat: expr
+        repeat-expr: num_id_entries
+    doc-ref: 'https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#resource-directory-table'
+
+  resource_directory_entry:
+    params:
+      - id: depth
+        type: u4
+    seq:
+      - id: resource_type
+        type: u4
+        enum: enum_resource_type
+        if: depth == 0
+      - id: ofs_name
+        type: u4
+        if: depth != 0
+      - id: ofs_data_entry
+        type: u4
```

Comment:
I think splitting these `u4` fields to `b1/b31` [bit-sized integers](https://doc.kaitai.io/user_guide.html#_bit_sized_integers) pairs would be a better idea than having this `& ~0x80000000` stuff everywhere.

```suggestion
      - id: is_name
        type: b1
        if: depth != 0
      - id: ofs_name
        type: b31
        if: depth != 0
      - id: is_data_entry
        type: b1
      - id: ofs_data_entry
        type: b31
```

Reply:
Good idea, thank you!